akka-actor/src/main/scala/akka/io/dns/CachePolicy.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.io.dns
+
+import akka.annotation.ApiMayChange
+
+import scala.concurrent.duration.{ Duration, FiniteDuration, _ }
+
+object CachePolicy {
+
+  @ApiMayChange
+  sealed trait CachePolicy
+  @ApiMayChange
+  case object Never extends CachePolicy
+  @ApiMayChange
+  case object Forever extends CachePolicy
+  @ApiMayChange
+  final class Ttl private (val value: FiniteDuration) extends CachePolicy {
+    if (value <= Duration.Zero) throw new IllegalArgumentException(s"TTL values must be a positive value.")
+    import akka.util.JavaDurationConverters._
+    def getValue: java.time.Duration = value.asJava
+
+    override def equals(other: Any): Boolean = other match {
+      case that: Ttl ⇒ value == that.value
+      case _         ⇒ false
+    }
+
+    override def hashCode(): Int = value.hashCode()
+
+    override def toString = s"Ttl($value)"
+  }
+  @ApiMayChange
+  object Ttl {
+    def unapply(ttl: Ttl): Option[FiniteDuration] = Some(ttl.value)
+    def fromPositive(value: FiniteDuration): Ttl = {
+      new Ttl(value)
+    }
```

Comment:
Where is the Java API?

Reply:
Well spotted, @hepin1989.

@ignasi35 add an overload which takes `java.time.Duration` and delegate to the scala impl by calling `.asScala` after file-wide importing `import akka.util.JavaDurationConverters._`.

Reply:
https://github.com/ignasi35/akka/pull/1

Reply:
Thanks for the callout @hepin1989 ! :-)
And thanks @dwijnand for the PR